templates/bucket-replication-lambda.yaml

```diff
@@ -0,0 +1,200 @@
+# This template creates an S3 bucket and a Lambda function
+#
+# The S3 bucket is granted permissions to be a Synapse storage location, *and* an S3 replication destination
+# from a collaborator's AWS account.
+#
+# The Lambda function is triggered by PUT requests to the bucket. It reads the files, computes and writes md5 sums
+# back to the bucket, and adds file handles to Synapse.
+#
+# This stack will need to be created with SynapseStorageLocationId = "" (blank). After the bucket is created,
+# the Synapse storage location must be created externally. Then update the stack with the new value for
+# SynapseStorageLocationId.
+#
+
+AWSTemplateFormatVersion: "2010-09-09"
+
+Parameters:
+  FunctionName:
+    Type: String
+    Description: Name of lambda function
+  FunctionCodeBucket:
+    Type: String
+    Description: S3 bucket where lambda deployment package is stored
+  FunctionCodeKey:
+    Type: String
+    Description: S3 key of lambda deployment package file
+  SynapseUsername:
+    Type: String
+    Description: Synapse username used to create files in Synapse.
+  SynapseApikey:
+    Type: String
+    Description: Synapse API key (associated with SynapseUsername) used to authenticate to Synapse.
+  SynapseStorageLocationId:
+    Type: Number
+    Description: >
+      Integer location ID for the bucket containing the files to be added to Synapse. Note a blank value should
+      be provided the when the stack is created.
+      See https://docs.synapse.org/articles/custom_storage_location.html
+  SynapseNotifyUsers:
+    Type: String
+    Description: >
+      Comma-delimited list of numeric Synapse user IDs to be notified when an attempt to sync to Synapse is made
+      by the Lambda function. Example: the `ownerId` returned by a call to the Synapse REST API /userProfile.
+  ReplicationSourceAccountId:
+    Type: Number
+    Description: ID of the AWS account holding the replication source bucket.
```

Comment:
wondering why this is necessary?  doesn't AWS supports bucket replication natively?  https://docs.aws.amazon.com/AmazonS3/latest/dev/replication.html

Reply:
The source bucket will be owned by a different AWS account (owned by a collaborator). So we need to explicitly give that account ID permission to replicate into ours.

Reply:
Changed the name to SourceAccountId.